src/cudamatrix/cu-matrix.cc

```diff
@@ -2528,8 +2528,45 @@ void CuMatrixBase<Real>::AddElements(Real alpha,
 }
 
 template<typename Real>
+void CuMatrixBase<Real>::AddElements(Real alpha, const CuArray<Int32Pair> &indexes,
+				     const Real *input) {
+  KALDI_ASSERT(input != NULL);
```

Comment:
Before checking that input != NULL, you might want to do
if (indexes.Dim() == 0)
  return;
We generally prefer all these routines to just return and do nothing for empty input, and having a NULL pointer would be fine if indexes.size() were empty.